recipes/newrelic/apm/php/debian.yml

```diff
@@ -533,7 +547,11 @@ install:
               # This ensures we always get the most up to date newrelic.ini file on the
               # system.
               #
-              mv ${ini_full_name} ${target_ini_dir}
+              ini_source_dir=$(dirname "$(realpath "$ini_full_name")")
+              ini_destination_dir=$(realpath "$target_ini_dir")
+              if [[ "$ini_source_dir" != "$ini_destination_dir" ]]; then
+                mv "${ini_full_name}" "${ini_destination_dir}"
+              fi
```

Comment:
Finally, this code block is a consequence of everything else described above. `$ini_full_name` holds the path to  `newrelic.ini` in `${ini}`. 

With all the context above, and the model used by the recipe to first install via the package-based installer and then use tarball, with the former leading to placing a `newrelic.ini` in `mods-available`, `target_ini_dir` was always meant to be `mods-available`, prior to these changes too (to move any updated `newrelic.ini` to replace the `newrelic.ini` in `mods-available`).

However, now with the concept of "updating" `newrelic.ini` in `mods-available` vanishing (if it already exists, no other `newrelic.ini` is added to other directories), and the probability of `${ini}` being `mods-available` when a `newrelic.ini` is found in `mods-available`, it is very likely that the path to `ini_full_name` and `target_ini_dir` would now be the same; in which case it wouldn't make sense to perform the move operation in the code block. Nevertheless, to make this check and only perform this when both of these paths are unequal, the condition has been added.

Reply:
The reason why this piece of code is necessary is that if a user installs the PHP agent manually using a tarball installer, they will have newrelic.ini in fpm or apache2 and cli, but not in mods-available as per agent logic. If, in the future, user want to run the guided install again on same host, the agent will not add newrelic.ini to any locations. However, the agent will inform us that newrelic.ini is already present in cli/conf.d and fpm/conf.d. Therefore, we will move the newrelic.ini to mods-available using this piece of code.